### PR TITLE
CORTX-33151 Codacy fixes related to style and security

### PR DESCRIPTION
**Pull request by seagate-sarang-sawant**
Problem: disable false positive, conflicting rules and reduce the Codacy issues.
Solution:
a) Fix all pep257 issues.
b) False positive and conflicting Codacy rules are disabled in codacy flake8.
c) Fix medium priority and security issues.

### CODE REVIEW

**Comment by d-nayak** [comment]
@seagate-sarang-sawant - Please handle DCO check - it's failing.

**Review by d-nayak — APPROVED** [review]

**Review by pavankrishnat — APPROVED** [review]

**Comment by pavankrishnat** [comment]
> @seagate-sarang-sawant - Please handle DCO check - it's failing.

yes, try edit commit message with signature

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -908,7 +908,7 @@ def list2dict(
     for node in nodes_data_hctl:
         node_svc_info: Dict[str, List[str]] = {}
         for service in node['svcs']:
-            if not service['name'] in node_svc_info.keys():
```
Comment (pavankrishnat) [inline provisioning/miniprov/hare_mp/main.py:911]: why to remove `.keys()`

Can you test manually this changes, as it not getting covered by any PR checks.

Comment (seagate-sarang-sawant) [reply provisioning/miniprov/hare_mp/main.py:911]: @pavankrishnat  the default search in dict is in keys and time complexity is O(1).  Converting into list every time for each service is O(n) operation.  We can add a test or test manually.

**Review by mssawant — APPROVED** [review]

**Review by Shreya-18 — APPROVED** [review]

**Comment by seagate-sarang-sawant** [comment]
retest this please

**Pull request merged**